src/cwt.rs

```diff
@@ -167,28 +151,32 @@ impl TryFrom<&[u8]> for Cwt {
             return Err(InvalidPartsCount(parts.len()));
         }
         let header_protected_raw = (parts[0].as_bytes().ok_or(HeaderNotBinary)?).clone();
-        let header_protected: CwtHeader =
-            ciborium::de::from_reader::<'_, Value, _>(header_protected_raw.as_slice())
-                .map_err(|_| HeaderNotValidCbor)?
-                .as_map()
-                .ok_or(HeaderNotMap)?
-                .as_slice()
-                .into();
+
         let header_unprotected = parts[1].clone();
         let payload_raw = (parts[2].as_bytes().ok_or(PayloadNotBinary)?).clone();
         let signature = (parts[3].as_bytes().ok_or(SignatureNotBinary)?).clone();
 
+        let unprotected_header_iter = parts[1].as_map().unwrap().iter();
```

Comment:
when does the unwrap() fail? you could do as you do in line 163 as well, I think.

Reply:
Good catch, let me review these nasty `unwrap()`s

Reply:
Addressed!